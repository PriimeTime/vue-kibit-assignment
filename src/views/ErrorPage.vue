<script setup lang="ts">
import HeaderText from "@/components/HeaderText.vue";
import DefaultLayout from "@/layouts/DefaultLayout.vue";
</script>

<template>
  <DefaultLayout>
    <div class="w-full h-full flex flex-col justify-center items-center mt-10">
      <HeaderText>Page not found.</HeaderText>
    </div>
  </DefaultLayout>
</template>
