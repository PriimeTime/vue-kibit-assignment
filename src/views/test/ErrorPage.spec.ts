import { describe, it, expect, vi } from "vitest";
import { mount } from "@vue/test-utils";
import ErrorPage from "@/views/ErrorPage.vue";
import { createTestingPinia } from "@pinia/testing";
import { createRouter, createWebHistory } from "vue-router";

const mockRouter = createRouter({
  history: createWebHistory(),
  routes: [{ path: "/", component: { template: "<div>Home</div>" } }],
});

describe("ErrorPage", () => {
  it("renders without crashing", () => {
    const wrapper = mount(ErrorPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("displays the error message", () => {
    const wrapper = mount(ErrorPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.text()).toContain("Page not found.");
  });

  it("matches snapshot", () => {
    const wrapper = mount(ErrorPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
