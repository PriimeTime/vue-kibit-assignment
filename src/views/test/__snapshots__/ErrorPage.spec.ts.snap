// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > matches snapshot 1`] = `
"<header class="flex h-16 bg-teal-900">
  <div class="flex items-center w-2/3 text-xl md:text-3xl whitespace-nowrap text-white p-2 hover:cursor-pointer"> Job Ad Plaform </div>
  <div class="flex justify-end w-1/3 p-2"><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Home</button>
    <!--v-if--><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Login</button>
  </div>
</header>
<main>
  <div class="w-full h-full flex flex-col justify-center items-center mt-10">
    <div class="text-3xl font-bold">Page not found.</div>
  </div>
</main>"
`;
